util/asm_dev/armv8/p256/src/main.c

```diff
@@ -0,0 +1,212 @@
+/*
+------------------------------------------------------------------------------------
+ Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ SPDX-License-Identifier: Apache-2.0
+------------------------------------------------------------------------------------
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include "p256.h"
+
+#ifdef SELECT_FN
+#define ENTRY_LONG_SIZE  12  //12 64-bit (long) words, 3 coordinates
+#define NUM_ENTRIES      16
+#define TABLE_LONG_SIZE (NUM_ENTRIES * ENTRY_LONG_SIZE)
+
+#define ENTRY_LONG_SIZE_W7  8   // 2 coordinates
+#define NUM_ENTRIES_W7     64
+#define TABLE_LONG_SIZE_W7 (NUM_ENTRIES_W7 * ENTRY_LONG_SIZE_W7)
+#endif
+
+#ifdef BEEU
+#define K_MAX   2000
+// P-256 group order in words (little-endian) obtained from debugging
+// ./third_party/boringssl/crypto/crypto_test --gtest_filter=P256_X86_64Test.*
+static const uint64_t order_words[4] =
+{
+    0xf3b9cac2fc632551, 0xbce6faada7179e84, 0xffffffffffffffff, 0xffffffff00000000
+};
+// In bytes from ec.c
+//{
+//    0xFF, 0xFF, 0xFF, 0xFF, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0xFF, 0xFF,
+//    0xFF, 0xFF, 0xFF, 0xFF, 0xBC, 0xE6, 0xFA, 0xAD, 0xA7, 0x17, 0x9E, 0x84,
+//    0xF3, 0xB9, 0xCA, 0xC2, 0xFC, 0x63, 0x25, 0x51,
+//};
+
+// expected inverse of a = 2, obtained from debugging as above
+static const uint64_t out_exp_2[4] =
+{
+    0x79dce5617e3192a9, 0xde737d56d38bcf42, 0x7fffffffffffffff, 0x7fffffff80000000
+};
+
+static void beeu_print(const int res, const uint64_t out_exp[4], const uint64_t out[4])
+{
+    uint32_t k;
+    if(res == 1)
+    {
+        for (k = 0; k < 4; k++)
+        {
+            if (out_exp[k] != out[k])
+            {
+                break;
+            }
+            printf("k:%d, out_exp: %lX, out: %lX\n", k, out_exp[k], out[k]);
+        }
+        printf("out_exp_p: %lX, out_p: %lX\n", (uintptr_t)out_exp, (uintptr_t)out);
+
+        if (4 == k)
+        {
+            printf("beeu Test passed\n");
+        }
+    }
+}
+#endif
+
+int main()
```

Comment:
Do we have similar tests for the production assembly? Or is that not possible to implement in our test framework?

Reply:
Yes, in `crypto/fipsmodule/ec/p256-nistz_test.cc`. The beeu tests are more extensive in that file since they use Montgomery multiplication to verify the output of multiplying a field member by its inverse. 
(The select tests were written before you pointed me to that file, Torben, and they turned out the same.)